README.md

```diff
@@ -1,65 +1,62 @@
-ROS2 for Rust
-=============
+# ROS2 for Rust
 
-Introduction
-------------
+## Overview
 
 This is a set of projects (bindings, code generator, examples and more) that enables developers to write ROS2
 applications in Rust.
 
-Features
---------
+## Status
+
+This project is still in early development stage. It's just a step toward a first class Rust support in ROS2.
+
+## Features
 
 The current set of features include:
 - Generation of all builtin ROS types
 - Support for publishers and subscriptions
 - Tunable QoS settings
 
-What's missing?
----------------
+## What's missing?
 
 Lots of things!
-- Nested types
 - An ament build type for Cargo. The current examples use CMake to install and build the binaries... and it's really ugly.
 - Component nodes
 - Clients and services
 - Tests
 - Documentation
 
-Sounds great, how can I try this out?
--------------------------------------
-
-The following steps show how to build the examples:
-
-```
-mkdir -p ~/ros2_rust_ws/src
-cd ~/ros2_rust_ws
-wget https://raw.githubusercontent.com/esteve/ros2_rust/master/ros2_rust.repos
-vcs import ~/ros2_rust_ws/src < ros2_rust.repos
-cd ~/ros2_rust_ws/src/ros2/rosidl_typesupport
-patch -p1 < ../../ros2_rust/ros2_rust/rosidl_typesupport_ros2_rust.patch
-cd ~/ros2_rust_ws
-src/ament/ament_tools/scripts/ament.py build --isolated
-```
+### Limitations
 
-Now you can just run a bunch of examples.
+- messages are deep-copied and this can be terribly inefficient for big messages like images; the current solution leverages C typesupport implementations and might benefits from a direct serialization/deserialization
+- the current solution for crates export with CMake is not very robust
+- `rclrs` interface is very limited for now and might not be so much idiomatic yet, any help and suggestion on the interface would be greatly appreciated
+- due to the current ROS2 support of non-default clients, packages containing definitions of messages used in Rust crates must be present in the current workspace; otherwise message crates generation won't be triggered
+- due to a not yet understood issue, `find_package` inclusion order matters in `CMakeLists.txt` of ROS packages containing message definitions:
+    `rosidl_default_generators` must be included before any other ROS packages containing message definitions, as in the following sample from `std_msgs`
+    ```cmake
+    find_package(rosidl_default_generators REQUIRED)
+    find_package(builtin_interfaces REQUIRED)
+    ```
 
-### Publisher and subscriber
+## Quickstart
 
-Publisher:
+### Build
 
+```bash
+source /opt/ros/crystal/setup.bash
+colcon build --symlink-install --merge-install --cmake-args -G Ninja
```

Comment:
There's no need for Ninja, the default builder should work. Have you tried this with an isolated workspace? (i.e. without --merge-install)